frontend-opal/src/components/PortfolioContainer/PortfolioContainer.js

```diff
@@ -1,46 +1,79 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { View } from 'react-native';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import thunks from 'thunks/portfolio';
+
 import styles from './styles';
 
 import ValueInfo from './ValueInfo';
 import AllocationInfo from './AllocationInfo';
 import Instruments from './Instruments';
 
-const PortfolioContainer = () => {
-  const totalInvestment = '3,400';
-  const pieData = [
-    { x: 'IT', y: 40, label: '40%' },
-    { x: 'Finance', y: 30, label: '30%' },
-    { x: 'Energy', y: 30, label: '30%' }
-  ];
-  const legendData = [
-    { name: 'IT', symbol: { type: 'square' } },
-    { name: 'Finance', symbol: { type: 'square' } },
-    { name: 'Energy', symbol: { type: 'square' } }
-  ];
-  const pieColor = ['#711702', '#A91600', '#E12C00'];
-  const instrumentsList = [
-    {
-      company: 'FB - Facebook Inc.',
-      stockExchange: 'NASDAQ',
-      /* eslint-disable-next-line */
-    profileImg: require('./Instruments/images/avatarPlaceholder.png'),
-      positions: 136,
-      marketValue: 11863.28,
-      unrlzedPLPercentage: '4.5%',
-      unrlzedPL: 510.86
-    }
-  ];
+const PortfolioContainer = ({ portfolio, onGetPortfolioData }) => {
+  useEffect(() => onGetPortfolioData(), [onGetPortfolioData]);
 
   return (
-    <View style={styles.portfolioArea}>
-      <ValueInfo totalInvestment={totalInvestment} />
+    <>
+      {portfolio[0] !== undefined && (
+        <View style={styles.portfolioArea}>
+          <ValueInfo totalInvestment={portfolio[0].totalInvestment} />
 
-      <AllocationInfo pieData={pieData} pieColor={pieColor} legendData={legendData} />
+          <AllocationInfo
+            pieData={portfolio[0].pieData}
+            pieColor={portfolio[0].pieColor}
+            legendData={portfolio[0].legendData}
+          />
 
-      <Instruments InstrumentList={instrumentsList} />
-    </View>
+          <Instruments instrumentList={portfolio[0].instrumentsList} />
+        </View>
+      )}
+    </>
   );
 };
 
-export default PortfolioContainer;
+const mapStateToProps = state => ({
+  portfolio: state.portfolio
+});
+
+const mapDispatchToProps = dispatch => {
+  return {
+    onGetPortfolioData: () => dispatch(thunks.getPortfolioData())
+  };
+};
+
+export default connect(mapStateToProps, mapDispatchToProps)(PortfolioContainer);
+
+PortfolioContainer.propTypes = {
```

Comment:
@WenyuD instead of passing down everything from PortfolioScreen, connect the separate components: ValueInfo, AllocationInfo, InstrumentList